### PR TITLE
LH tests

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

**Comment by fisx** [comment]
`retryWhileN` wouldn't do it (I think), since that doesn't catch exceptions.

File: services/galley/test/integration/API/Teams/LegalHold.hs [hunk 1]
```diff
@@ -607,7 +613,9 @@ withTestService mkApp go = do
     srv <- liftIO . Async.async $
         Warp.runTLS tlss defs $
             mkApp buf
-    go buf `finally` liftIO (Async.cancel srv)
+    recovering def [\_ -> Handler $ \(SomeException _) -> pure False]
```
Comment (ChrisPenner) [inline services/galley/test/integration/API/Teams/LegalHold.hs]: I'm a little confused, this says DON'T retry on ANY exceptions... not positive, but I think this does basically nothing?

I think we just want `recoverAll`?

**Comment by ChrisPenner** [comment]
The default retry strategy seems to retry 5 times with only a 50ms delay between; that's only 250ms of waiting total; not sure how long it takes k8s to catch up; but this seems a bit short?

Maybe exponential backoff instead?

```haskell
myBackoff = exponentialBackoff 50 <> limitRetries 5
```

**Comment by neongreen** [comment]
I looked at the PR title and thought those would be LiquidHaskell tests :(

File: services/galley/test/integration/API/Teams/LegalHold.hs (hunk 1 above)
Comment (fisx) [reply services/galley/test/integration/API/Teams/LegalHold.hs]: you're not nearly as confused as i am.  :)

also you're right.  fixing...

**Pull request merged**